### ISSUE TITLE
False positives for Style/SymbolArray in literal symbol array

### ISSUE THREAD

**Issue by lostie** [rubocop/rubocop#12071]
## Expected behavior

- The `Style/SymbolArray` should not detect an offence when using square brackets in a symbol literal
- The `Style/SymbolArray` should not detect an offence when using interpolation in a symbol literal

## Actual behavior

Rubocop detect an `Style/SymbolArray` offence

## Steps to reproduce the problem

```ruby
%i[foo[4] bar[4]] # <= detects a `Style/SymbolArray` offence

%I[foo_#{index}] # <= detects a `Style/SymbolArray` offence
```

## RuboCop version

```
$ bundle exec rubocop -V
1.55.0 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.2.2) [arm64-darwin22]
  - rubocop-factory_bot 2.23.1
  - rubocop-graphql 1.3.0
  - rubocop-performance 1.18.0
  - rubocop-rails 2.20.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.22.0
```
